### PR TITLE
Schedulerepair

### PR DESCRIPTION
**Pull request by kylealexanderhackney**
## Describe this PR

Adding schedule repair endpoint to repairs api supportting schedule repair under hact

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: RepairsApi/V2/Controllers/RepairsController.cs [hunk 1]
```diff
@@ -49,7 +49,28 @@ public async Task<IActionResult> RaiseRepair([FromBody] RaiseRepair request)
...
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("schedule")]
+        public async Task<IActionResult> ScheduleRepair([FromBody] ScheduleRepair request)
+        {
+            try
+            {
+                var result = await _createWorkOrderUseCase.Execute(request.ToDb());
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Controllers/RepairsController.cs:73]: Add tests for new exception

File: RepairsApi/V2/Factories/DBModelFactory.cs [hunk 2]
```diff
@@ -28,6 +51,178 @@ public static WorkOrder ToDb(this Generated.RaiseRepair raiseRepair)
...
+                    ElevationReferenceSystem = request.ElevationReferenceSystem.First(),
+                    Latitude = double.Parse(request.Latitude.First()),
+                    Longitude = double.Parse(request.Longitude.First()),
+                    PositionalAccuracy = request.PositionalAccuracy,
+                    Polyline = request.Polyline.MapList(p => p.ToDb())
+                };
+            }
+            catch (FormatException e)
+            {
+                throw new NotSupportedException(Resources.InvalidGeographicalLocation, e);
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Factories/DBModelFactory.cs:168]: probably also needs a test

File: RepairsApi.Tests/V2/Controllers/RepairsControllerTests.cs [hunk 3]
```diff
@@ -51,7 +51,7 @@ public void SetUp()
         private void ConfigureGenerator()
         {
             _generator = new Generator<WorkOrder>()
-                .AddDefaultValueGenerators();
+                .AddWorkOrderGenerators();
         }
```
Comment (kylealexanderhackney) [inline RepairsApi.Tests/V2/Controllers/RepairsControllerTests.cs:56]: Add unit test for new endpoint

File: RepairsApi/V2/Factories/DBModelFactory.cs [hunk 4]
```diff
@@ -28,6 +28,178 @@ public static WorkOrder ToDb(this Generated.RaiseRepair raiseRepair)
             };
         }
 
+        public static WorkOrder ToDb(this Generated.ScheduleRepair raiseRepair)
+        {
+            return new WorkOrder
+            {
+                DescriptionOfWork = raiseRepair.DescriptionOfWork,
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Factories/DBModelFactory.cs:35]: Null propagation on nested toDb calls

File: RepairsApi/V2/Factories/DBModelFactory.cs [hunk 5]
```diff
@@ -28,6 +28,178 @@ public static WorkOrder ToDb(this Generated.RaiseRepair raiseRepair)
...
+                {
+                    Elevation = double.Parse(request.Elevation.First()),
+                    ElevationReferenceSystem = request.ElevationReferenceSystem.First(),
+                    Latitude = double.Parse(request.Latitude.First()),
+                    Longitude = double.Parse(request.Longitude.First()),
+                    PositionalAccuracy = request.PositionalAccuracy,
+                    Polyline = request.Polyline.MapList(p => p.ToDb())
+                };
+            }
+            catch (FormatException e)
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Factories/DBModelFactory.cs:166]: test for this exception

**Review by josh-dadswell-hackney — CHANGES_REQUESTED** [review]

**Review by josh-dadswell-hackney — APPROVED** [review]

**Review by alex-demetriou — APPROVED** [review]

**Pull request merged**